jl4-core/src/L4/Parser.hs

```diff
@@ -1009,11 +1009,17 @@ list = do
       <$  annoLexeme (spacedToken_ TKList)
       <*> annoHole (lsepBy (indentedExpr current) (spacedToken_ TComma))
 
-numericLit :: Parser Lit
-numericLit =
+intLit :: Parser Lit
+intLit =
   attachAnno $
     NumericLit emptyAnno
-      <$> annoEpa (spacedToken (fmap snd <$> preview #_TIntLit) "Numeric Literal")
+      <$> annoEpa (spacedToken (fmap (toRational . snd) <$> preview #_TIntLit) "Numeric Literal")
+
+floatLit :: Parser Lit
```

Comment:
the naming is a bit off. this is not a floating point number.

Reply:
Is it not? `1.1234567` is a floating point number, afaict.

Reply:
it's just a decimal number, from the user's point of view. a floating point number is when you can "move" the number via the exponent, e.g. IEEE754 floating point numbers